File: dot-net-projects-master/csasp-lms/UniversityLMS/Filters/RAuthorizeAttribute.cs
using System;
using System.Web.Mvc;

namespace UniversityLMS.Filters
{
    public class RAuthorizeAttribute: ActionFilterAttribute
    {

        public override void OnActionExecuting(ActionExecutingContext filterContext)
        {
            base.OnActionExecuting(filterContext);

            bool authorized = Convert.ToBoolean(filterContext.HttpContext.Session["UserIsAuthorized"]);

            if ((!authorized))
            {
                // Not logged in so send user to the login page
                filterContext.HttpContext.Response.Redirect("/Login");
            }
            else
            {
                filterContext.Controller.TempData["Username"] = filterContext.HttpContext.Session["Username"];
            }
        }
    }

}

File: dot-net-projects-master/vbasp-project-log/VBAspProjectLog/Filters/RAuthorizeAttribute.vb
Imports System.Web.Mvc

Public Class RAuthorizeAttribute

    Inherits ActionFilterAttribute

    Public Overrides Sub OnActionExecuting(filterContext As ActionExecutingContext)
        MyBase.OnActionExecuting(filterContext)

        Dim authorized As Boolean = Convert.ToBoolean( _
            filterContext.HttpContext.Session("UserIsAuthorized"))

        If (Not authorized) Then
            ' Not logged in so send user to the login page
            filterContext.HttpContext.Response.Redirect("/Login")
        Else
            filterContext.Controller.TempData("Username") = _
                filterContext.HttpContext.Session("Username")
        End If

    End Sub


End Class
